src/gui/SubWindow.cpp

```diff
@@ -220,34 +239,36 @@ void SubWindow::resizeEvent( QResizeEvent * event )
 	{
 		buttonBarWidth = buttonBarWidth + m_buttonSize.width() + buttonGap;
 		m_maximizeBtn->move( middleButtonPos );
-		m_maximizeBtn->show();
+		m_restoreBtn->move( middleButtonPos );
+		if( isMaximized() )
+		{
+			m_restoreBtn->show();
```

Comment:
Perhaps... 

```cpp
m_restoreBtn->setVisible( isMaximized() );
m_maximizeBtn->setVisible( !isMaximized() );
```

May also remove the need to hide everything at the top of the function.

Just a side note... I'm not sure how well the optimizers handle this type of logic, but down the road `isMaximized()` may benefit from being cached into a variable for the resize event if we ever are in a scenario where we do a bunch of window resizing (such as on project load, etc).  Minor point though.... otherwise, the wasted CPU cycles should be relatively benign.